node/core/approval-voting/src/lib.rs

```diff
@@ -683,9 +690,21 @@ where
 
 	let mut last_finalized_height: Option<BlockNumber> = None;
 
+	let startup_delay =
+		|| async move { future::pending::<()>().timeout(STARTUP_WAIT_DURATION).await };
+
 	loop {
 		let mut overlayed_db = OverlayedBackend::new(&backend);
 		let actions = futures::select! {
+			_res = startup_delay().fuse() => {
+				if !state.recovery_state.complete() {
+					let height = last_finalized_height.unwrap_or_default();
+					let actions = handle_startup(&mut overlayed_db, &mut state).await?;
+					actions.into_iter().filter_map(|(candidate_height, a)| if candidate_height > height { Some(a) } else { None }).collect()
```

Comment:
Good, I was just looking for this check.

Reply:
I have modified the PR using the FusedStream. I will burn in on Rococo and then it should be good to go.